applications/FluidDynamicsApplication/custom_elements/two_fluid_navier_stokes.cpp

```diff
@@ -1879,15 +1926,240 @@ void TwoFluidNavierStokes<TElementData>::ComputeSplitting(
+{
+    VectorType rhs = ZeroVector(NumNodes*NumNodes); //Size is NumNodes x (NumDim+1)
+
+    for (unsigned int intgp = 0; intgp < rInterfaceWeights.size(); ++intgp){
+        for (unsigned int j = 0; j < NumNodes; ++j){
+            for (unsigned int dim = 0; dim < NumNodes-1; ++dim){
+                rhs[ j*(NumNodes) + dim ] -= coefficient*(rInterfaceNormalsNeg[intgp])[dim]
+                    *rCurvature(intgp)*rInterfaceWeights(intgp)*rInterfaceShapeFunctions(intgp,j);
+            }
+        }
+    }
+
+    noalias(rRHS) += rhs;
+}
+
+template <class TElementData>
+void TwoFluidNavierStokes<TElementData>::PressureGradientStabilization(
+    TElementData& rData,
+    const Vector& rInterfaceWeights,
+    const Matrix& rEnrInterfaceShapeFunctionPos,
+    const Matrix& rEnrInterfaceShapeFunctionNeg,
+    const GeometryType::ShapeFunctionsGradientsType& rInterfaceShapeDerivatives,
+    MatrixType& rKeeTot,
+    VectorType& rRHSeeTot)
+{
+    MatrixType kee = ZeroMatrix(NumNodes, NumNodes);
+    VectorType rhs_enr = ZeroVector(NumNodes);
+
+    Matrix enr_neg_interp = ZeroMatrix(NumNodes, NumNodes);
+    Matrix enr_pos_interp = ZeroMatrix(NumNodes, NumNodes);
+
+    double positive_density = 0.0;
+    double negative_density = 0.0;
+    double positive_viscosity = 0.0;
+    double negative_viscosity = 0.0;
+
+    for (unsigned int i = 0; i < NumNodes; ++i){
+        if (rData.Distance[i] > 0.0){
+            enr_neg_interp(i, i) = 1.0;
+            positive_density = rData.NodalDensity[i];
+            positive_viscosity = rData.NodalDynamicViscosity[i];
+        } else{
+            enr_pos_interp(i, i) = 1.0;
+            negative_density = rData.NodalDensity[i];
+            negative_viscosity = rData.NodalDynamicViscosity[i];
+        }
+    }
+
+    GeometryType::ShapeFunctionsGradientsType EnrichedInterfaceShapeDerivativesPos = rInterfaceShapeDerivatives;
+    GeometryType::ShapeFunctionsGradientsType EnrichedInterfaceShapeDerivativesNeg = rInterfaceShapeDerivatives;
+
+    for (unsigned int i = 0; i < rInterfaceShapeDerivatives.size(); ++i){
+        EnrichedInterfaceShapeDerivativesPos[i] = prod(enr_pos_interp, rInterfaceShapeDerivatives[i]);
+    }
+
+    for (unsigned int i = 0; i < rInterfaceShapeDerivatives.size(); ++i){
+        EnrichedInterfaceShapeDerivativesNeg[i] = prod(enr_neg_interp, rInterfaceShapeDerivatives[i]);
+    }
+
+    double positive_volume = 0.0;
+    double negative_volume = 0.0;
+    for (unsigned int igauss_pos = 0; igauss_pos < rData.w_gauss_pos_side.size(); ++igauss_pos){
+        positive_volume += rData.w_gauss_pos_side[igauss_pos];
+    }
+
+    for (unsigned int igauss_neg = 0; igauss_neg < rData.w_gauss_neg_side.size(); ++igauss_neg){
+        negative_volume += rData.w_gauss_neg_side[igauss_neg];
+    }
+    const double element_volume = positive_volume + negative_volume;
+
+    auto geom = this->GetGeometry();
+    const double h_elem = ElementSizeCalculator<Dim,NumNodes>::AverageElementSize(geom);
```

Comment:
```suggestion
    const double h_elem = rData.ElementSize;
```
Indeed, you already have the element size in the data structure, so you avoid computing it twice.

Reply:
Done.